routes/applications.rb

```diff
@@ -36,10 +36,18 @@ def self.registered(app)
             json data
           end
 
-          get_application_waiver = lambda do
+          get_application_and_waiver = lambda do
+            application = WRNGApplication.find(params[:id])
             waiver = Waiver.find_by(application: params[:id])
-            if waiver
-              json(waiver)
+            if application && waiver
+              if application && (application[:type] == 'camper' || application[:type] == 'volunteer')
```

Comment:
You don't need to check for application here, you already did that on ln. 42.

Reply:
Very true. Good point!